### PR TITLE
Add * as text auto in gitattributes

### PR DESCRIPTION
**Pull request by weshaggard**
@heaths testing out this change can you help me understand what negative impacts it has?

### CODE REVIEW

**Comment by heaths** [comment]
Opening and saving any file - even without explicit changes - will change the line endings on Windows (and probably MacOS). Everything will show up in a diff, and in my case for *.ps1 (after some were already committed) git returned error 128 in the AzDO builds for all platforms. Changing line endings after the fact (which is effectively what this will do) is often disasterous, which is why Git recommends (but, to this day, still doesn't do by default) to add a .gitattributes with `* text=auto` to your initial commit.

**Comment by heaths** [comment]
If you want an example, open eng\New-TestResources.ps1, save, notice there's a change, commit, and push to this PR. Kick off a build.

**Comment by weshaggard** [comment]
I don't get any diffs when I open a ps1 file locally. What is your core.autocrlf set to? Mine is set to true on my windows machine and I'm not getting any odd line ending diffs.

**Comment by weshaggard** [comment]
@heaths I still haven't seen any negative affects of this change is there something I'm missing?

**Comment by heaths** [comment]
Did you try modifying some text files that weren't treated as text before (e.g. .ps1 files). Their line endings change, so a complete diff will occur for every file touched.

**Comment by weshaggard** [comment]
Yes I've touched only as well as made small edits to a ps1 file and everything looks as it should to me.

File: .gitattributes [hunk 1]
```diff
@@ -1,3 +1,5 @@
+* text=auto
```
Comment (heaths) [inline .gitattributes]: Just to clean up this file a bit, should the plain text entries below (ones without a diff clause) be removed?

**Review by heaths — APPROVED** [review]

File: .gitattributes (hunk 1 above)
Comment (weshaggard) [reply .gitattributes]: Yeah we probably should removed the redundancy.

**Pull request merged**